File: rei-do-chopp-site/src/app/shared/models/requests/pagination.request.ts
import { OrdenationTypeEnum } from "../../enums/ordenation-type.enum";

export class PaginationRequest
{
  Page: number;
  PageSize: number;
  OrdenationField: string;
  OrdenationType: OrdenationTypeEnum

  constructor(params: Partial<PaginationRequest>)
  {
    this.Page = params.Page || 1;
    this.PageSize = params.PageSize || 10;
    this.OrdenationField = params.OrdenationField || "Id";
    this.OrdenationType = params.OrdenationType ?? OrdenationTypeEnum.Descendent;
  }
}

File: rei-do-chopp-api/ReiDoChopp.DataTransfer/Pagination/Requests/PaginationRequest.cs
using ReiDoChopp.Domain.Utils.Enums;

namespace ReiDoChopp.DataTransfer.Pagination.Requests
{
    public class PaginationRequest
    {
        private int pageSize;
        public int Page { get; set; }
        public int PageSize {
            get => pageSize;
            set => pageSize = (value < 100 ? value : 100);
        }
        public string OrdenationField { get; set; }
        public OrdenationTypeEnum OrdenationType { get; set; }

        public PaginationRequest(int pageSize, string ordenationField, OrdenationTypeEnum ordenationType)
        {
            Page = 1;
            PageSize = pageSize;
            OrdenationField = ordenationField;
            OrdenationType = ordenationType;
        }
    }
}
